pkg/instrumentation/dotnet.go

```diff
@@ -116,3 +112,25 @@ func injectDotNetSDK(logger logr.Logger, dotNetSpec v1alpha1.DotNet, pod corev1.
 	pod.Spec.Containers[index] = container
 	return pod
 }
+
+func trySetEnvVar(logger logr.Logger, container *corev1.Container, envVarName string, envVarValue string, concatValues bool) bool {
+	idx := getIndexOfEnv(container.Env, envVarName)
+	if idx < 0 {
+		container.Env = append(container.Env, corev1.EnvVar{
+			Name:  envVarName,
+			Value: envVarValue,
+		})
+		return true
+	}
+
+	if container.Env[idx].ValueFrom != nil {
+		// TODO add to status object or submit it as an event
+		logger.Info("Skipping DotNet SDK injection, the container defines env var value via ValueFrom", "envVar", envVarName, "container", container.Name)
```

Comment:
I think this behavior and the log message are misleading.

When `false` is returned then it interrupts processing next env vars.
It could leave the container in a "semi-instrumented" state. Is it expected?

This comment applies also to the code BEFORE the changes introduced in this PR. Therefore, it would be better to make a separate PR to address it.